script/whitelist/delegationFaucet/DelegationFaucet.sol

```diff
@@ -0,0 +1,189 @@
+     * DEFAULT_AMOUNT if _depositAmount is 0. Then Staker contract deposits into the strategy and delegates to operator.
+     * @param _operator The operator to delegate to
+     * @param _depositAmount The amount to deposit into the strategy
+     */
+    function mintDepositAndDelegate(address _operator, uint256 _depositAmount) public onlyOwner {
+        // Operator must be registered
+        require(delegation.isOperator(_operator), "DelegationFaucet: Operator not registered");
+        address staker = getStaker(_operator);
+        // Set deposit amount
+        if (_depositAmount == 0) {
+            _depositAmount = DEFAULT_AMOUNT;
+        }
+        stakeToken.mint(staker, _depositAmount);
+
+        // Deploy staker address if not already deployed, staker constructor will approve the StrategyManager to spend the stakeToken
+        if (!Address.isContract(staker)) {
+            Create2.deploy(
+                0,
+                bytes32(uint256(uint160(_operator))),
+                abi.encodePacked(type(DelegationFaucetStaker).creationCode, abi.encode(strategyManager, stakeToken))
+            );
+        }
+
+        // deposit into stakeToken strategy
+        uint256 shares = abi.decode(depositIntoStrategy(staker, stakeStrategy, stakeToken, _depositAmount), (uint256));
+        // increaseDelegatedShares if staker contract already delegated, o/w delegateTo operator
+        if (delegation.isDelegated(staker)) {
+            delegation.increaseDelegatedShares(staker, stakeStrategy, shares);
+        } else {
+            IDelegationManager.SignatureWithExpiry memory signatureWithExpiry;
+            delegateTo(_operator, signatureWithExpiry, bytes32(0));
+        }
+    }
+
+    function getStaker(address operator) public view returns (address) {
+        return
+            Create2.computeAddress(
+                bytes32(uint256(uint160(operator))), //salt
+                keccak256(
+                    abi.encodePacked(type(DelegationFaucetStaker).creationCode, abi.encode(strategyManager, stakeToken))
+                )
+            );
+    }
+
+    function depositIntoStrategy(
+        address staker,
+        IStrategy strategy,
+        IERC20 token,
+        uint256 amount
+    ) public onlyOwner returns (bytes memory) {
+        bytes memory data = abi.encodeWithSelector(
+            IStrategyManager.depositIntoStrategy.selector,
+            strategy,
+            token,
+            amount
+        );
+
+        return Staker(staker).callAddress(address(strategyManager), data);
+    }
+
+    function increaseDelegatedShares(
+        address staker,
+        IStrategy strategy,
+        uint256 shares
+    ) public onlyOwner returns (bytes memory) {
+        bytes memory data = abi.encodeWithSelector(
+            IDelegationManager.increaseDelegatedShares.selector,
+            strategy,
+            shares
+        );
+
+        return Staker(staker).callAddress(address(delegation), data);
+    }
```

Comment:
as noted above, this should always revert. if the intention here is to add shares to a staker, we'd need to mint more tokens and have the staker deposit them.

Reply:
This function is not needed and won't work anyways, adding more shares to a staker will just be called by the same function `mintDepositAndDelegate()`